vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+        name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?"))
+    ```
+
+-   Also, we need to mention in the `DetailsList()` function inside UI to only display the names we just assigned in the `columns` object.
+    Here is how the `ui` function now gets modified -
+
+    ``` r
+    #' @export
+    ui <- function(id) {
+      ns <- NS(id)
+        fluentPage(
+            Text("A randomly generated dataset of imaginary sales deals"),
+            DetailsList(items = shiny.fluent::fluentSalesDeals, columns = columns)
+        )
+    }
+    ```
+
+    Remember to include `tibble` inside `box::use()`, else the code will not be able to use the package.
+
+-   Upon running the app with all these changes, you should see the improved table -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_3.png")
+```
+
+#### Render the Table from the server side inside the `datatable` module
+
+-   Till now, we did not do anything inside the server side of the `app/view/datatable.R` file.
+    Let us now use the server side to render the table rather than the UI side.
+    This is particularly helpful when we want to introduce more features inside the table - such as user inputs.
+
+-   So, here is the new modified `app/view/databale.R` file
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS, uiOutput, renderUI],
+      shiny.fluent[fluentPage, Text, DetailsList],
+      tibble[tibble]
+    )
+
+    columns <- tibble(
+        fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+        name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?"))
+
+    #' @export
+    ui <- function(id) {
+      ns <- NS(id)
+        fluentPage(
+            Text("A randomly generated dataset of imaginary sales deals"),
+            uiOutput(ns("datatable")))
+    }
+
+    #' @export
+    server <- function(id) {
+      moduleServer(id, function(input, output, session) {
+        output$datatable <- renderUI({
+            DetailsList(items = shiny.fluent::fluentSalesDeals, columns = columns)
+        })
+      })
+    }
+    ```
+
+The **`DetailsList`** table is being rendered inside the **`renderUI`** function, which generates UI output dynamically based on the **`items`** and **`columns`** arguments.
+The **`renderUI`** function is triggered whenever there is a change in the input values or reactive expressions that depend on them (we will explore this further).
+
+The **`columns`** data frame defines the field names and column names for the **`DetailsList`**.
+The **`fieldName`** column contains the names of the columns in the **`shiny.fluent::fluentSalesDeals`** data frame, and the **`name`** column contains the display names for those columns.
+
+The **`ns`** variable is used to generate namespace for the **`uiOutput`** function.
+This is important because it ensures that the input and output elements in the UI do not clash with elements from other modules or UI components in the same application.
```

Comment:
This is a repetition, you already wrote about modules.

Reply:
removed this explanation as it is already discussed earlier